### PR TITLE
add ruby lambda layer draft

### PR DESCRIPTION
**Pull request by xuan-cao-swi**

### CODE REVIEW

File: ruby/sample-apps/function/Gemfile
Comment (jerrytfleung) [inline ruby/sample-apps/function/Gemfile:1]: Is it intended to have an empty file?

File: ruby/sample-apps/function/lambda_function.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require 'json'
+require 'opentelemetry-sdk'
+
+def lambda_handler(event:, context:)
+  if defined?(::OpenTelemetry::SDK)
+    { statusCode: 200, body: "Hello #{::OpenTelemetry::SDK::VERSION}" }
+  else
+    { statusCode: 200, body: "Missing OpenTelemetry" }
+  end
+end
```
Comment (jerrytfleung) [inline ruby/sample-apps/function/lambda_function.rb]: nits

File: ruby/sample-apps/template.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        - AmazonS3ReadOnlyAccess
+      Layers:
+        - !Ref OTelLayer
+      Events:
+        getEndpoint:
+          Type: Api
+          Properties:
+            RestApiId: !Ref api
+            Path: /
+            Method: GET
```
Comment (jerrytfleung) [inline ruby/sample-apps/template.yml]: nits

File: ruby/src/zip_ruby_layer.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+set -e
+set -u
+
+echo_usage() {
+	echo "usage: Build Lambda layer/application by SAM"
+	echo " -n <specify layer name>"
+}
+
+is_sample() {
```
Comment (jerrytfleung) [inline ruby/src/zip_ruby_layer.sh]: do we need this function? 
Seem it is not referenced in this bash

File: ruby/sample-apps/function/Gemfile
Comment (xuan-cao-swi) [reply ruby/sample-apps/function/Gemfile:1]: Should not be empty!

File: ruby/src/zip_ruby_layer.sh (hunk 3 above)
Comment (xuan-cao-swi) [reply ruby/src/zip_ruby_layer.sh]: no, should just remove it.

File: ruby/sample-apps/function/lambda_function.rb (hunk 1 above)
Comment (xuan-cao-swi) [reply ruby/sample-apps/function/lambda_function.rb]: any suggestion on nits?

File: ruby/sample-apps/template.yml (hunk 2 above)
Comment (xuan-cao-swi) [reply ruby/sample-apps/template.yml]: any suggestion on nits?

File: ruby/sample-apps/function/lambda_function.rb (hunk 1 above)
Comment (jerrytfleung) [reply ruby/sample-apps/function/lambda_function.rb]: Need a newline at the end of file

File: ruby/sample-apps/template.yml (hunk 2 above)
Comment (jerrytfleung) [reply ruby/sample-apps/template.yml]: Need a newline at the end of file

**Comment by jerrytfleung** [comment]
Do we need to add instruction in the README.md that how to set the AWS_LAMBDA_EXEC_WRAPPER. 

like what is it in nodejs https://github.com/open-telemetry/opentelemetry-lambda/tree/main/nodejs#opentelemetry-lambda-nodejs
```
Layer for running NodeJS applications on AWS Lambda with OpenTelemetry. Adding the layer and pointing to it with the AWS_LAMBDA_EXEC_WRAPPER environment variable will initialize OpenTelemetry, enabling tracing with no code change.

To use, add the layer to your function configuration and then set AWS_LAMBDA_EXEC_WRAPPER to /opt/otel-handler.
```

**Comment by jerrytfleung** [comment]
Basically it is looking good to me. Do you have a lambda setup in our AWS organization so I can see the ruby lambda layer and collector lambda layer installed and working fine?

**Comment by xuan-cao-swi** [comment]
> Basically it is looking good to me. Do you have a lambda setup in our AWS organization so I can see the ruby lambda layer and collector lambda layer installed and working fine?

Hi @jerrytfleung , yes, I have one function with the layer [here](https://us-east-1.console.aws.amazon.com/lambda/home?region=us-east-1#/functions/xuan-cao-ruby-test?tab=code)

Although I don't have the collector lambda layer attached to the lambda function. However, I did couple tests that directly send the trace to our swo backend.

> Do we need to add instruction in the README.md that how to set the AWS_LAMBDA_EXEC_WRAPPER.

I agree and I will add them.

Correct me if I am wrong. I think `AWS_LAMBDA_EXEC_WRAPPER` (or aws lambda wrapper) runs user-provided script or binary during the lambda initialization (before reach to lambda function), and I saw most script from nodejs and python are related to setting the default environmental variables for opentelemetry. So probably future contributor can help decide which default variable is suitable for ruby agent.

**Comment by xuan-cao-swi** [comment]
@jerrytfleung I have updated the readme.

**Review by jerrytfleung — APPROVED** [review]
LGTM

**Pull request closed without merging**